SeaPublicWebsite/Controllers/EnergyEfficiencyController.cs

```diff
@@ -257,10 +256,12 @@ public async Task<ViewResult> ConfirmAddress_Get(string reference, string postco
         {
             List<EpcSearchResult> epcSearchResults = await epcApi.GetEpcsInformationForPostcodeAndBuildingNameOrNumber(postcode, number);
             
-            var filteredEpcSearchResults = epcSearchResults.Where(e =>
+            var matchingEpcSearchResults = epcSearchResults.Where(e =>
                 e.Address1.Contains(number, StringComparison.OrdinalIgnoreCase) ||
                 e.Address2.Contains(number, StringComparison.OrdinalIgnoreCase)).ToList();
 
+            var filteredEpcSearchResults = await FilterExpiredEpc(matchingEpcSearchResults);
```

Comment:
Perhaps I'm misunderstanding, but is there a possible simpler version of this that just does the filtering on the results as they already are, without additional database checks? What's the reason for not just picking out unique addresses from the `matchingEpcSearchResults` variable at this stage?

Reply:
I've had a bit of a poke around and from what I can tell, the endpoint for searching by address returns a less detailed Epc object which doesn't include the 'IsLatestAssessmentForAddress' property, whereas the Epc by id endpoint does return this property.